src/init.c

```diff
@@ -1263,11 +1263,17 @@ void jl_get_builtin_hooks(void)
     jl_interrupt_exception = jl_new_struct_uninit((jl_datatype_t*)core("InterruptException"));
     jl_boundserror_type    = (jl_datatype_t*)core("BoundsError");
     jl_memory_exception    = jl_new_struct_uninit((jl_datatype_t*)core("MemoryError"));
+    #ifdef _OS_LINUX_
+    #ifdef _P32
+        jl_float64_type->alignment = 4;
```

Comment:
Not sure if it's different now, but back then the alignment would be set to 8, since it's an 8 byte object but on Linux 32 bit the ABI specifies 4 byte alignment.